pkg/apiserver/internal/v1/account.go

```diff
@@ -0,0 +1,172 @@
+/*
+Copyright 2020 The KubeCarrier Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	"context"
+	"fmt"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"k8s.io/apimachinery/pkg/labels"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	catalogv1alpha1 "github.com/kubermatic/kubecarrier/pkg/apis/catalog/v1alpha1"
+	v1 "github.com/kubermatic/kubecarrier/pkg/apiserver/api/v1"
+	"github.com/kubermatic/kubecarrier/pkg/apiserver/internal/oidc"
+	"github.com/kubermatic/kubecarrier/pkg/apiserver/internal/util"
+)
+
+type accountServer struct {
+	client client.Client
+}
+
+var _ v1.AccountServiceServer = (*accountServer)(nil)
+
+// +kubebuilder:rbac:groups=catalog.kubecarrier.io,resources=accounts,verbs=get;list;watch
+
+func NewAccountServiceServer(c client.Client) v1.AccountServiceServer {
+	return &accountServer{
+		client: c,
+	}
+}
+
+func (o accountServer) List(ctx context.Context, req *v1.AccountListRequest) (res *v1.AccountList, err error) {
```

Comment:
we can't test this method as for now, because we can ExtractUser from context, one of the solutions can be to add use dependency injection and pass `oidc` as a dependency to `NewAccountServiceServer` and then we can mock it in the tests

Reply:
We are currently unit-testing the `handleListRequest`, I think it should be enough, for List functionality, we are testing it in e2e tests

Reply:
ok for this PR, but if we will have more and more of such places, we need to think about solution